toolchain/check/import_cpp.cpp

```diff
@@ -351,53 +353,67 @@ static auto MapType(Context& context, clang::QualType type) -> TypeExpr {
 // declaration. If the function declaration has no parameters, it returns
 // `SemIR::InstBlockId::Empty`. In the case of an unsupported parameter type, it
 // returns `SemIR::InstBlockId::None`.
+// TODO: Consider refactoring to extract and reuse more logic from
+// `HandleAnyBindingPattern()`.
 static auto MakeParamPatternsBlockId(Context& context, SemIR::LocId loc_id,
                                      const clang::FunctionDecl& clang_decl)
     -> SemIR::InstBlockId {
   if (clang_decl.parameters().empty()) {
     return SemIR::InstBlockId::Empty;
   }
-  SemIR::CallParamIndex next_index(0);
   llvm::SmallVector<SemIR::InstId> params;
   params.reserve(clang_decl.parameters().size());
   for (const clang::ParmVarDecl* param : clang_decl.parameters()) {
     clang::QualType param_type = param->getType().getCanonicalType();
-    SemIR::TypeId type_id =
-        GetPatternType(context, MapType(context, param_type).type_id);
+
+    // Mark the start of a region of insts, needed for the type expression
+    // created later with the call of `EndSubpatternAsExpr()`.
+    BeginSubpattern(context);
+    auto [type_inst_id, type_id] = MapType(context, param_type);
+    // Type expression of the binding pattern - a single-entry/single-exit
+    // region that allows control flow in the type expression e.g. fn F(x: if C
+    // then i32 else i64).
+    SemIR::ExprRegionId type_expr_region_id =
+        EndSubpatternAsExpr(context, type_inst_id);
+
     if (type_id == SemIR::ErrorInst::TypeId) {
       context.TODO(loc_id, llvm::formatv("Unsupported: parameter type: {0}",
                                          param_type.getAsString()));
       return SemIR::InstBlockId::None;
     }
+
     llvm::StringRef param_name = param->getName();
-    SemIR::EntityNameId entity_name_id = context.entity_names().Add(
-        {.name_id =
-             (param_name.empty())
-                 // Translate an unnamed parameter to an underscore to
-                 // match Carbon's naming of unnamed/unused function params.
-                 ? SemIR::NameId::Underscore
-                 : SemIR::NameId::ForIdentifier(
-                       context.sem_ir().identifiers().Add(param_name)),
-         .parent_scope_id = SemIR::NameScopeId::None});
-    SemIR::InstId binding_pattern_id = AddInstInNoBlock(
+    SemIR::NameId name_id =
+        (param_name.empty())
```

Comment:
Remove extra parenthesis.
```suggestion
        param_name.empty()
```

Reply:
Done.